minari/utils.py

```diff
@@ -353,8 +353,8 @@ def get_average_reference_score(
 
 def create_dataset_from_buffers(
     dataset_id: str,
-    env: gym.Env,
     buffer: List[Dict[str, Union[list, Dict]]],
+    env: Optional[gym.Env] = None,
```

Comment:
notice that this is a breaking change (positional calling will not work anymore), we may want to delay this for minari 0.5, that may be the next release anyway

Reply:
Okay, will work on the complexity (if conditions). 
Yes, it is a breaking change. I will keep the changes ready in any case.

I also had a question regarding the test case:

Which of the following options do we use for the env buffers in the absence of an env?
a. any buffers already available. I don't know if there is any, do you?
b. create buffer data by running one of the gym env
c. Manually create a dummy env buffer object

Reply:
What do you mean with env buffers? We have a field in the metadata to save the env_spec; in case env is missing we will miss that attribute in the metadata. The env can missing, for example if the data comes from a real robot with no associated simulated environment. Does this answer your question?

Reply:
Let me rephrase my question. What do we pass as observation space and action space when env is None for the test case? Which one if those 3 options listed would be better?

Reply:
If env is not specified, observation_space and action_space must be specified
This is what the if branch of lines 436-440 does, right?

Reply:
The user must specify it

Reply:
Yes, the user must specify it.
That's why I was wondering what to use for the "test case". I wanted to check if we already have some action space observation space data for the test case, or should I create one. I listed those 3 options to get your advice on what would be the best option.

Reply:
Oh clear now, you are asking it for tests
For that you can follow the same workflow of this: https://github.com/Farama-Foundation/Minari/blob/c43a61249ac51dd663a12983b7050aced97115b0/tests/utils/test_dataset_creation.py#L112

So creating the buffer from env (they are test env testing different spaces; I don't think it is necessary to test that function, one is enough). If you need to copy-paste the code from that test, feel free to create a method in tests/common.py with a function to create the buffer from an env

Reply:
Yes, thanks a lot, I was asking about the test. In the absence of env, we need to pass action space or observation space mandatorily. Hence, I wanted ideas on what arguments to pass for observation space and action space parameters when we test this scenario.

Reply:
You should use the same observation_space and action_space of the environment that generates the dataset (`env.observation_space` and env.action_space`), then, for the sake of the test, you don't pass the environment but only these spaces

Reply:
Have resolved the complexity for if while raising error and suppressed pyright errors. Working on the test case.

Reply:
Have resolved the complexity for if while raising error and suppressed pyright errors. Working on the test case.